<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="LucianoRogerio" />

<meta name="date" content="2022-01-26" />

<title>IITA Breeding Scheme</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CassavaBreedingSim</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="VD_Optimization.html">VDs Simulations</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Simulations Scripts
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="SingularityIm.html">Singularity Image</a>
    </li>
    <li>
      <a href="BurnIn_IITA.html">IITA BurnIn Breeding Scheme</a>
    </li>
    <li>
      <a href="IITABS.html">IITA Breeding Scheme</a>
    </li>
    <li>
      <a href="NVDP1.html">IITA NVDP1</a>
    </li>
    <li>
      <a href="NVD3.html">IITA NVDP2</a>
    </li>
    <li>
      <a href="1-3YrsIITABS.html">1-3 Years Candidate</a>
    </li>
  </ul>
</li>
<li>
  <a href="graphics.html">Figures</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LucianoRogerio/CassavaBreedingSim">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">IITA Breeding Scheme</h1>
<h4 class="author">LucianoRogerio</h4>
<h4 class="date">2022-01-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CassavaBreedingSim/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210916code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210916code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210916)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLucianoRogerioCassavaBreedingSimtreee9e11cdec5f778982d9bcab7d2720d18ca3455d2targetblanke9e11cda"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/LucianoRogerio/CassavaBreedingSim/tree/e9e11cdec5f778982d9bcab7d2720d18ca3455d2" target="_blank">e9e11cd</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomLucianoRogerioCassavaBreedingSimtreee9e11cdec5f778982d9bcab7d2720d18ca3455d2targetblanke9e11cda" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/LucianoRogerio/CassavaBreedingSim/tree/e9e11cdec5f778982d9bcab7d2720d18ca3455d2" target="_blank">e9e11cd</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/.DS_Store
    Ignored:    output/.Rhistory
    Ignored:    output/DataSimPlotIITA1_3_Par.rds

Unstaged changes:
    Modified:   analysis/IITABS.Rmd
    Modified:   analysis/graphics.Rmd
    Modified:   output/DataSimPlotIITA1_3_Acc.rds
    Modified:   output/DataSimPlotIITA1_3_F1.rds
    Modified:   output/DataSimPlotIITA1_3_UYT2.rds
    Deleted:    output/ParentOriginGS_1Yr.csv
    Deleted:    output/ParentOriginGS_2Yrs.csv
    Deleted:    output/ParentOriginGS_3Yrs.csv
    Deleted:    output/ParentOriginPS.csv
    Deleted:    output/ParentOriginPSPlus.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/IITABS.Rmd</code>) and HTML (<code>docs/IITABS.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/e9e11cdec5f778982d9bcab7d2720d18ca3455d2/analysis/IITABS.Rmd" target="_blank">e9e11cd</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-04-01
</td>
<td>
Updates on the layout of the figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/a294aaea68c6186e87ce40605c40c0a975afa5cb/analysis/IITABS.Rmd" target="_blank">a294aae</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-03-18
</td>
<td>
New results with time cycle fixed, with selection of 30 parents, and prediction of the selection accuracy for Clones at CET
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/e3304e973c084523deefdf7923d75453b2ba295b/analysis/IITABS.Rmd" target="_blank">e3304e9</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-03-10
</td>
<td>
Update result with Crossing block taking one year to obtain seeds
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/3a65e6fc233298dc3214d7e178959267c2d5d1d3/analysis/IITABS.Rmd" target="_blank">3a65e6f</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-03-04
</td>
<td>
Updates first week of March
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LucianoRogerio/CassavaBreedingSim/3a65e6fc233298dc3214d7e178959267c2d5d1d3/docs/IITABS.html" target="_blank">3a65e6f</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-03-04
</td>
<td>
Updates first week of March
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/7f2d4afdc82da3a8717052b9a34c01a721409677/analysis/IITABS.Rmd" target="_blank">7f2d4af</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-02-18
</td>
<td>
Updates Results IITA Simulations
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/727209036522a3cc612c39ac31693230cd758c9b/analysis/IITABS.Rmd" target="_blank">7272090</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-02-11
</td>
<td>
Update of results Feb 10
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LucianoRogerio/CassavaBreedingSim/727209036522a3cc612c39ac31693230cd758c9b/docs/IITABS.html" target="_blank">7272090</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-02-11
</td>
<td>
Update of results Feb 10
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/fed39b3b030a3d0ee517371897327254fd9c7f9b/analysis/IITABS.Rmd" target="_blank">fed39b3</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-02-10
</td>
<td>
Add Burn in Web site, Origin of the parents, new results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LucianoRogerio/CassavaBreedingSim/fed39b3b030a3d0ee517371897327254fd9c7f9b/docs/IITABS.html" target="_blank">fed39b3</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-02-10
</td>
<td>
Add Burn in Web site, Origin of the parents, new results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/a977b0f85caa50ba3d6ab7b4800badcbf6e65c57/analysis/IITABS.Rmd" target="_blank">a977b0f</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-02-03
</td>
<td>
Results IITA BreedSchem 1-3 Years
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LucianoRogerio/CassavaBreedingSim/a977b0f85caa50ba3d6ab7b4800badcbf6e65c57/docs/IITABS.html" target="_blank">a977b0f</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-02-03
</td>
<td>
Results IITA BreedSchem 1-3 Years
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/93d96138687aaa1ee86970a0293a97dfa7d9aa11/analysis/IITABS.Rmd" target="_blank">93d9613</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-01-27
</td>
<td>
New Results from first Simulations with NVDP
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LucianoRogerio/CassavaBreedingSim/93d96138687aaa1ee86970a0293a97dfa7d9aa11/docs/IITABS.html" target="_blank">93d9613</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-01-27
</td>
<td>
New Results from first Simulations with NVDP
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LucianoRogerio/CassavaBreedingSim/fb5483506c7ab6e9d56990ff02b90df05d6ae77a/docs/IITABS.html" target="_blank">fb54835</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-01-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/bfd89848c2a3aac1cdb902f8b85903184111ee78/analysis/IITABS.Rmd" target="_blank">bfd8984</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-01-27
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LucianoRogerio/CassavaBreedingSim/69e0b17c8fecab6a8e2e527fef47b02ec89d6cb1/docs/IITABS.html" target="_blank">69e0b17</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-01-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/CassavaBreedingSim/blob/99148164874d9ab03e76350bb0f13578e0c4935a/analysis/IITABS.Rmd" target="_blank">9914816</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-01-26
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="simulations-with-the-iita-breeding-scheme" class="section level2">
<h2>Simulations with the IITA Breeding Scheme</h2>
<p><br /></p>
<table>
<thead>
<tr class="header">
<th align="center">FunctionsArg</th>
<th align="center">FunctionName</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">selCritPop</td>
<td align="center">parentSelCritBLUP</td>
</tr>
<tr class="even">
<td align="center">selCritPipe</td>
<td align="center">productSelCritBLUP</td>
</tr>
<tr class="odd">
<td align="center">productFunc</td>
<td align="center">productPipeline</td>
</tr>
<tr class="even">
<td align="center">popImprovFunc</td>
<td align="center">popImprovByParentSel</td>
</tr>
<tr class="odd">
<td align="center"></td>
<td align="center"></td>
</tr>
</tbody>
</table>
<p><br /></p>
<div id="open-r-inside-a-singularity-container" class="section level4">
<h4>Open R inside a singularity container</h4>
<pre class="bash"><code>screen -R LucianoSim;

# as I runned on Robbins Server I got to ask access to slurm

salloc -n 8 --mem 80G;

cd /workdir/lbraatz

singularity shell rocker2.sif

R</code></pre>
</div>
<div id="run-r-inside-a-singularity-container" class="section level4">
<h4>Run R inside a singularity container</h4>
<ul>
<li>IITA PS - cbsu lm32 - screen IITAPS</li>
</ul>
<pre class="r"><code>.libPaths(c(.libPaths(), &quot;/programs/R-4.1.2/library/&quot;, &quot;/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/&quot;))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select &lt;- dplyr::select
mutate &lt;- dplyr::mutate

schemeDF &lt;- read.csv(here::here(&quot;data&quot;,&quot;baselineScheme - IITA.csv&quot;), 
                     header = T, stringsAsFactors = F, sep = &quot;;&quot;)

bsp1 &lt;- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 8, nYrsAsCandidates = 3, maxTrainingPopSize = 6620,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = &quot;AYT&quot;,  RmStagePhen = &quot;SDN&quot;,
                   nParents = 30, nCrosses = 400, nProgeny = 30, nClonesToNCRP = 3,
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = T,
                   nCyclesToKeepRecords = 4,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here(&quot;code&quot;,&quot;runSchemesPostBurnIn.R&quot;))

nRepT &lt;- 40#readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% nrow()
NCT &lt;- RhpcBLASctl::get_num_procs()
nRepMap &lt;- 8
reps &lt;- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]

for(i in 1:max(reps)) {

burnIn_IITA_PS &lt;- readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% .[reps==i,]
start &lt;- proc.time()[3]
postBurnIn_IITA_PS &lt;- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = &quot;parentSelCritBLUP&quot;,
                                           selCritPipe = &quot;productSelCritBLUP&quot;,
                                           productFunc = &quot;productPipelinePostBurnIn&quot;,
                                           popImprovFunc = &quot;popImprovByParentSel&quot;,
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end &lt;- proc.time()[3]; print(paste0(floor((end - start)/60), &quot;mins &quot;, floor(((end - start)%%60)), &quot;s elapsed - bspP&quot;))

RdsName &lt;- paste(&quot;postBurnIn_IITA_PS&quot;, i, &quot;.rds&quot;, sep = &quot;&quot;)
saveRDS(postBurnIn_IITA_PS,file = here::here(&quot;output&quot;, RdsName))
rm(postBurnIn_IITA_PS); rm(RdsName); rm(burnIn_IITA_PS)
}

xx &lt;- readRDS(here::here(&quot;output&quot;, &quot;postBurnIn_IITA_PS1.rds&quot;))

xxx &lt;- xx$SimOutput[[1]]$records$PopImprov$grmData[[27]] %&gt;% filter(pop ==&quot;c&quot;)

table(xxx$stage)
sum(is.na(xxx$gebv))</code></pre>
<pre><code> ****** 1 to 8

 ****** 9 to 16
 [1] &quot;42mins 24s elapsed - bspP&quot;

 ****** 17 to 24
 [1] &quot;42mins 5s elapsed - bspP&quot;

 ****** 25 to 32
 [1] &quot;42mins 30s elapsed - bspP&quot;

 ****** 33 to 40
 [1] &quot;43mins 8s elapsed - bspP&quot;</code></pre>
<p>joint all the Results in a same rds File</p>
<pre class="r"><code>suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(80)
files &lt;- here::here(&quot;output&quot;, dir(&quot;output&quot;)[grep(pattern = &quot;postBurnIn_IITA_PS*&quot;, dir(&quot;output&quot;))])
SimResults &lt;- NULL
for(i in files){
  Results &lt;- readRDS(i)
  SimResults &lt;- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = here::here(&quot;output&quot;, &quot;postBurnIn_IITA_PS.rds&quot;))
rm(SimResults)</code></pre>
</div>
<div id="obtain-the-ratio-of-parent-selected-per-stage" class="section level3">
<h3>Obtain the ratio of Parent selected per stage</h3>
<pre class="r"><code>library(tidyverse); library(AlphaSimR)
library(here)
schemeDF &lt;- read.csv(here::here(&quot;data&quot;,&quot;baselineScheme - IITA.csv&quot;), 
                     header = T, stringsAsFactors = F, sep = &quot;;&quot;)
simulations &lt;- readRDS(here::here(&quot;output&quot;, &quot;postBurnIn_IITA_PS.rds&quot;))

RelPar &lt;- tibble()
for(repl in 1:nrow(SimResults)){
  for(YEAR in length(SimResults$SimOutput[[repl]]$records$CrossingBlock$grmData)){
RelPar &lt;- rbind(RelPar,
                SimResults$SimOutput[[repl]]$records$CrossingBlock$grmData[[YEAR]][SimResults$SimOutput[[repl]]$records$CrossingBlock$grmData[[YEAR]]$pop == &quot;c&quot;,])
}
}

write.table(x = RelPar, file = here::here(&quot;output&quot;, &quot;ParentOriginPS.csv&quot;),
            sep = &quot;,&quot;, quote = FALSE,
            row.names = FALSE, col.names = TRUE)</code></pre>
<pre class="r"><code>.libPaths(c(.libPaths(), &quot;/programs/R-4.1.2/library/&quot;, &quot;/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/&quot;))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select &lt;- dplyr::select
mutate &lt;- dplyr::mutate

schemeDF &lt;- read.csv(here::here(&quot;data&quot;,&quot;baselineScheme - IITAPSExtra.csv&quot;), 
                     header = T, stringsAsFactors = F, sep = &quot;;&quot;)

bsp1 &lt;- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 8, nYrsAsCandidates = 3, maxTrainingPopSize = 6620,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = &quot;AYT&quot;,  RmStagePhen = &quot;SDN&quot;,
                   nParents = 30, nCrosses = 400, nProgeny = 30, nClonesToNCRP = 3,
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = T,
                   nCyclesToKeepRecords = 4,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here(&quot;code&quot;,&quot;runSchemesPostBurnIn.R&quot;))

nRepT &lt;- 40#readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% nrow()
NCT &lt;- RhpcBLASctl::get_num_procs()
nRepMap &lt;- 8
reps &lt;- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]


set.seed(501)

for(i in 1:max(reps)) {

burnIn_IITA_PS &lt;- readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% .[reps==i,]
start &lt;- proc.time()[3]
postBurnIn_IITA_PS &lt;- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = &quot;parentSelCritBLUP&quot;,
                                           selCritPipe = &quot;productSelCritBLUP&quot;,
                                           productFunc = &quot;productPipelinePostBurnIn&quot;,
                                           popImprovFunc = &quot;popImprovByParentSel&quot;,
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end &lt;- proc.time()[3]; print(paste0(floor((end - start)/60), &quot;mins &quot;, floor(((end - start)%%60)), &quot;s elapsed - bspP&quot;))

RdsName &lt;- paste(&quot;postBurnIn_IITA_PSPlus&quot;, i, &quot;.rds&quot;, sep = &quot;&quot;)
saveRDS(postBurnIn_IITA_PS,file = here::here(&quot;output&quot;, RdsName))
rm(postBurnIn_IITA_PS); rm(RdsName); rm(burnIn_IITA_PS)
}</code></pre>
<pre><code> ****** 1 to 8
 [1] &quot;39mins 39s elapsed - bspP&quot;

 ****** 9 to 16
 [1] &quot;40mins 33s elapsed - bspP&quot;

 ****** 17 to 24
 [1] &quot;40mins 22s elapsed - bspP&quot;

 ****** 25 to 32

 ****** 33 to 40
 [1] &quot;47mins 3s elapsed - bspP&quot;
 </code></pre>
<p>joint all the Results in a same rds File</p>
<pre class="r"><code>suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(80)
files &lt;- here::here(&quot;output&quot;, dir(&quot;output&quot;)[grep(pattern = &quot;postBurnIn_IITA_PSPlus*&quot;, dir(&quot;output&quot;))])
SimResults &lt;- NULL
for(i in files){
  Results &lt;- readRDS(i)
  SimResults &lt;- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = here::here(&quot;output&quot;, &quot;postBurnIn_IITA_PSPlus.rds&quot;))
rm(SimResults)</code></pre>
<div id="with-genomic-selection" class="section level4">
<h4>With Genomic Selection</h4>
<p><br /></p>
<table>
<thead>
<tr class="header">
<th align="center">FunctionsArg</th>
<th align="center">FunctionName</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">selCritPop</td>
<td align="center">parentSelCritGEBV</td>
</tr>
<tr class="even">
<td align="center">selCritPipe</td>
<td align="center">productSelCritBLUP</td>
</tr>
<tr class="odd">
<td align="center">productFunc</td>
<td align="center">productPipeline</td>
</tr>
<tr class="even">
<td align="center">popImprovFunc</td>
<td align="center">popImprovByParentSel</td>
</tr>
<tr class="odd">
<td align="center"></td>
<td align="center"></td>
</tr>
</tbody>
</table>
<div id="parameters-changuing-in-different-scenarios" class="section level6">
<h6>Parameters Changuing in different scenarios</h6>
<ul>
<li>3 Yrs as Candidates - cbsu lm34 - screen BurnIn</li>
</ul>
<p><br /></p>
<pre class="r"><code>.libPaths(c(.libPaths(), &quot;/programs/R-4.1.2/library/&quot;, &quot;/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/&quot;))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select &lt;- dplyr::select
mutate &lt;- dplyr::mutate
AlphaSimHlpR::popImprovOutput

schemeDF &lt;- read.csv(here::here(&quot;data&quot;,&quot;baselineScheme - IITA.csv&quot;), 
                     header = T, stringsAsFactors = F, sep = &quot;;&quot;)

bsp1 &lt;- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 5, nYrsAsCandidates = 3, maxTrainingPopSize = 6620,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = &quot;CET&quot;,
                   RmStagePhen = &quot;SDN&quot;,
                   nParents = 30, nCrosses = 400, nProgeny = 30, nClonesToNCRP = 3,
                   parentsFlowering = 100,
                   TrainPopSel = &quot;trainPopSelNew&quot;,
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = F,
                   nCyclesToKeepRecords = 5,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here(&quot;code&quot;,&quot;runSchemesPostBurnIn.R&quot;))

nRepT &lt;- 40#readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% nrow()
NCT &lt;- RhpcBLASctl::get_num_procs()
nRepMap &lt;- 5
reps &lt;- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]

for(i in 1:ceiling(nRepT/nRepMap)) {

burnIn_IITA_PS &lt;- readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% .[reps==i,]
start &lt;- proc.time()[3]
postBurnIn_IITA_GS &lt;- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = &quot;parentSelCritGEBV&quot;,
                                           selCritPipe = &quot;productSelCritBLUP&quot;,
                                           productFunc = &quot;productPipelinePostBurnIn&quot;,
                                           popImprovFunc = &quot;popImprovByParentSel&quot;,
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end &lt;- proc.time()[3]; print(paste0(floor((end - start)/60), &quot;mins &quot;, floor(((end - start)%%60)), &quot;s elapsed - bspP&quot;))

RdsName &lt;- paste(&quot;postBurnIn_IITA_GS_3Yrs&quot;, i, &quot;.rds&quot;, sep = &quot;&quot;)
saveRDS(postBurnIn_IITA_GS,file = here::here(&quot;output&quot;, RdsName))
rm(postBurnIn_IITA_GS); rm(RdsName); rm(burnIn_IITA_PS)
}</code></pre>
<pre><code> ****** 1 to 5
 [1] &quot;650mins 3s elapsed - bspP&quot;
 
 ****** 6 to 10
 [1] &quot;327mins 59s elapsed - bspP&quot;

 ****** 11 to 15
 [1] &quot;366mins 4s elapsed - bspP&quot;
  
 ****** 20 to 25
 [1] &quot;371mins 9s elapsed - bspP&quot;
 
 ****** 26 to 30
 [1] &quot;333mins 40s elapsed - bspP&quot;
 
 ****** 31 to 35
 [1] &quot;366mins 29s elapsed - bspP&quot;
 
 ****** 36 to 40
 [1] &quot;362mins 18s elapsed - bspP&quot;
</code></pre>
<p>joint all the Results in a same rds File</p>
<pre class="r"><code>suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(60)
files &lt;- here::here(&quot;output&quot;, dir(&quot;output&quot;)[grep(pattern = &quot;postBurnIn_IITA_GS_3Yrs*&quot;, dir(&quot;output&quot;))])
SimResults &lt;- NULL
for(i in files){
  Results &lt;- readRDS(i)
  SimResults &lt;- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = &quot;output/postBurnIn_IITA_GS_3Yrs.rds&quot;)
rm(SimResults)</code></pre>
<ul>
<li>2 Yrs as Candidate - CBSU lm35 - screen IITA3Yr</li>
</ul>
<p><br /></p>
<pre class="r"><code>.libPaths(c(.libPaths(), &quot;/programs/R-4.1.2/library/&quot;, &quot;/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/&quot;))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select &lt;- dplyr::select
mutate &lt;- dplyr::mutate

schemeDF &lt;- read.csv(here::here(&quot;data&quot;,&quot;baselineScheme - IITA.csv&quot;), 
                     header = T, stringsAsFactors = F, sep = &quot;;&quot;)

bsp1 &lt;- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 5, nYrsAsCandidates = 2, maxTrainingPopSize = 6700,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = &quot;CET&quot;,
                   RmStagePhen = &quot;SDN&quot;,
                   nParents = 30, nCrosses = 400, nProgeny = 30, nClonesToNCRP = 3,
                   parentsFlowering = 100,
                   TrainPopSel = &quot;trainPopSelNew&quot;,
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = F,
                   nCyclesToKeepRecords = 5,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here(&quot;code&quot;,&quot;runSchemesPostBurnIn.R&quot;))

nRepT &lt;- 40#readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% nrow()
NCT &lt;- RhpcBLASctl::get_num_procs()
nRepMap &lt;- 8
reps &lt;- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]

for(i in 5:ceiling(nRepT/nRepMap)) {

burnIn_IITA_PS &lt;- readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% .[reps==i,]
start &lt;- proc.time()[3]
postBurnIn_IITA_GS &lt;- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = &quot;parentSelCritGEBV&quot;,
                                           selCritPipe = &quot;productSelCritBLUP&quot;,
                                           productFunc = &quot;productPipelinePostBurnIn&quot;,
                                           popImprovFunc = &quot;popImprovByParentSel&quot;,
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end &lt;- proc.time()[3]; print(paste0(floor((end - start)/60), &quot;mins &quot;, floor(((end - start)%%60)), &quot;s elapsed - bspP&quot;))

RdsName &lt;- paste(&quot;postBurnIn_IITA_GS_2Yrs&quot;, i, &quot;.rds&quot;, sep = &quot;&quot;)
saveRDS(postBurnIn_IITA_GS,file = here::here(&quot;output&quot;, RdsName))
rm(postBurnIn_IITA_GS); rm(RdsName); rm(burnIn_IITA_PS)
}</code></pre>
<pre><code> ****** 1 to 8
  [1] &quot;557mins 43s elapsed - bspP&quot;

 ****** 9 to 16
  [1] &quot;526mins 55s elapsed - bspP&quot;
  
 ****** 17 to 24
 1] &quot;544mins 32s elapsed - bspP&quot;
  
 ****** 25 to 32
 [1] &quot;575mins 20s elapsed - bspP&quot;

 ****** 33 to 40
 [1] &quot;576mins 29s elapsed - bspP&quot;
</code></pre>
<p>joint all the Results in a same rds File</p>
<pre class="r"><code>suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(60)
files &lt;- here::here(&quot;output&quot;, dir(&quot;output&quot;)[grep(pattern = &quot;postBurnIn_IITA_GS_2Yrs*&quot;, dir(&quot;output&quot;))])
SimResults &lt;- NULL
for(i in files){
  Results &lt;- readRDS(i)
  SimResults &lt;- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = &quot;output/postBurnIn_IITA_GS_2Yrs.rds&quot;)
rm(SimResults)</code></pre>
<ul>
<li>1 Yrs as Candidate - CBSU lm30 - screen IITA1Yr</li>
</ul>
<p><br /></p>
<pre class="r"><code>.libPaths(c(.libPaths(), &quot;/programs/R-4.1.2/library/&quot;, &quot;/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/&quot;))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select &lt;- dplyr::select
mutate &lt;- dplyr::mutate

schemeDF &lt;- read.csv(here::here(&quot;data&quot;,&quot;baselineScheme - IITA.csv&quot;), 
                     header = T, stringsAsFactors = F, sep = &quot;;&quot;)

bsp1 &lt;- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 5, nYrsAsCandidates = 1, maxTrainingPopSize = 7500,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = &quot;CET&quot;,
                   RmStagePhen = &quot;SDN&quot;,
                   nParents = 30, nCrosses = 400, nProgeny = 30, nClonesToNCRP = 3,
                   parentsFlowering = 100,
                   TrainPopSel = &quot;trainPopSelNew&quot;,
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = F,
                   nCyclesToKeepRecords = 5,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here(&quot;code&quot;,&quot;runSchemesPostBurnIn.R&quot;))

nRepT &lt;- 40#readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% nrow()
NCT &lt;- RhpcBLASctl::get_num_procs()
nRepMap &lt;- 8
reps &lt;- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]

for(i in 4:ceiling(nRepT/nRepMap)) {

burnIn_IITA_PS &lt;- readRDS(here::here(&quot;output&quot;, &quot;BurnIn_IITA.rds&quot;)) %&gt;% .[reps==i,]
start &lt;- proc.time()[3]
postBurnIn_IITA_GS &lt;- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = &quot;parentSelCritGEBV&quot;,
                                           selCritPipe = &quot;productSelCritBLUP&quot;,
                                           productFunc = &quot;productPipelinePostBurnIn&quot;,
                                           popImprovFunc = &quot;popImprovByParentSel&quot;,
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end &lt;- proc.time()[3]; print(paste0(floor((end - start)/60), &quot;mins &quot;, floor(((end - start)%%60)), &quot;s elapsed - bspP&quot;))

RdsName &lt;- paste(&quot;postBurnIn_IITA_GS_1Yrs&quot;, i, &quot;.rds&quot;, sep = &quot;&quot;)
saveRDS(postBurnIn_IITA_GS,file = here::here(&quot;output&quot;, RdsName))
rm(postBurnIn_IITA_GS); rm(RdsName); rm(burnIn_IITA_PS)
}</code></pre>
<pre><code> ****** 1 to 8
  [1] &quot;383mins 9s elapsed - bspP&quot;

 ****** 9 to 16
 [1] &quot;359mins 5s elapsed - bspP&quot;
 
 ****** 17 to 24
 [1] &quot;375mins 35s elapsed - bspP&quot;
  
 ****** 25 to 32
 [1] &quot;323mins 7s elapsed - bspP&quot;
 
 ****** 33 to 40
 [1] &quot;380mins 24s elapsed - bspP&quot;</code></pre>
<p>joint all the Results in a same rds File</p>
<pre class="r"><code>suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(80)
files &lt;- here::here(&quot;output&quot;, dir(&quot;output&quot;)[grep(pattern = &quot;postBurnIn_IITA_GS_1Yrs*&quot;, dir(&quot;output&quot;))])
SimResults &lt;- NULL
for(i in files){
  Results &lt;- readRDS(i)
  SimResults &lt;- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = &quot;output/postBurnIn_IITA_GS_1Yr.rds&quot;)</code></pre>
<p><a href="VD_Optimization.html">back</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Big Sur 11.6.5

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     rstudioapi_0.13  whisker_0.4      knitr_1.38      
 [5] magrittr_2.0.3   workflowr_1.7.0  R6_2.5.1         rlang_1.0.2     
 [9] fastmap_1.1.0    fansi_1.0.3      stringr_1.4.0    tools_4.1.2     
[13] xfun_0.30        utf8_1.2.2       cli_3.2.0        git2r_0.30.1    
[17] jquerylib_0.1.4  htmltools_0.5.2  ellipsis_0.3.2   rprojroot_2.0.3 
[21] yaml_2.3.5       digest_0.6.29    tibble_3.1.6     lifecycle_1.0.1 
[25] crayon_1.5.1     later_1.3.0      sass_0.4.1       vctrs_0.4.0     
[29] promises_1.2.0.1 fs_1.5.2         glue_1.6.2       evaluate_0.15   
[33] rmarkdown_2.13   stringi_1.7.6    bslib_0.3.1      compiler_4.1.2  
[37] pillar_1.7.0     jsonlite_1.8.0   httpuv_1.6.5     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
