---
title: "IITA Breeding Scheme"
author: "LucianoRogerio"
date: "2022-01-26"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


## Simulations with the IITA Breeding Scheme

<br />

| FunctionsArg  | FunctionName         |
| :-----------: | :------------------: |
| selCritPop    | parentSelCritBLUP    |
| selCritPipe   | productSelCritBLUP   |
| productFunc   | productPipeline      |
| popImprovFunc | popImprovByParentSel |
|               |                      |

<br />

#### Open R inside a singularity container

```{bash Open singularity, eval = F}
screen -R LucianoSim;

# as I runned on Robbins Server I got to ask access to slurm

salloc -n 8 --mem 80G;

cd /workdir/lbraatz

singularity shell rocker2.sif

R
```


#### Run R inside a singularity container

- IITA PS - cbsu lm32 - screen IITAPS

```{r PostBurnIn IITA PS, eval = F}

.libPaths(c(.libPaths(), "/programs/R-4.1.2/library/", "/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/"))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select <- dplyr::select
mutate <- dplyr::mutate

schemeDF <- read.csv(here::here("data","baselineScheme - IITA.csv"), 
                     header = T, stringsAsFactors = F, sep = ";")

bsp1 <- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 4, nYrsAsCandidates = 3, maxTrainingPopSize = 6620,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = "AYT",  RmStagePhen = "SDN",
                   nParents = 30, nCrosses = 400, nProgeny = 35, nClonesToNCRP = 3,
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = T,
                   nCyclesToKeepRecords = 4,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here("code","runSchemesPostBurnIn.R"))

nRepT <- 40#readRDS(here::here("output", "BurnIn_IITA.rds")) %>% nrow()
NCT <- RhpcBLASctl::get_num_procs()
nRepMap <- 8
reps <- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]

for(i in 1:max(reps)) {

burnIn_IITA_PS <- readRDS(here::here("output", "BurnIn_IITA.rds")) %>% .[reps==i,]
start <- proc.time()[3]
postBurnIn_IITA_PS <- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = "parentSelCritBLUP",
                                           selCritPipe = "productSelCritBLUP",
                                           productFunc = "productPipelinePostBurnIn",
                                           popImprovFunc = "popImprovByParentSel",
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end <- proc.time()[3]; print(paste0(floor((end - start)/60), "mins ", floor(((end - start)%%60)), "s elapsed - bspP"))

RdsName <- paste("postBurnIn_IITA_PS", i, ".rds", sep = "")
saveRDS(postBurnIn_IITA_PS,file = here::here("output", RdsName))
rm(postBurnIn_IITA_PS); rm(RdsName); rm(burnIn_IITA_PS)
}

xx <- readRDS(here::here("output", "postBurnIn_IITA_PS1.rds"))

xxx <- xx$SimOutput[[1]]$records$CrossingBlock$grmData[[27]] %>% filter(pop =="c")

table(xxx$stage)
sum(is.na(xxx$gebv))
```

```
 ****** 1 to 8
 [1] "33mins 1s elapsed - bspP"

 ****** 9 to 16
 [1] "33mins 16s elapsed - bspP"

 ****** 17 to 24

 ****** 25 to 32
 [1] "32mins 59s elapsed - bspP"

 ****** 33 to 40
 [1] "32mins 26s elapsed - bspP"
```


joint all the Results in a same rds File
```{r Joint PostBurnIn IITA PS data, eval = FALSE}
suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(80)
files <- here::here("output", dir("output")[grep(pattern = "postBurnIn_IITA_PS*", dir("output"))])
SimResults <- NULL
for(i in files){
  Results <- readRDS(i)
  SimResults <- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = here::here("output", "postBurnIn_IITA_PS.rds"))
rm(SimResults)
```

### Obtain the ratio of Parent selected per stage

```{r Parent Selection origin PS, eval = FALSE}
library(tidyverse); library(AlphaSimR)
library(here)
schemeDF <- read.csv(here::here("data","baselineScheme - IITA.csv"), 
                     header = T, stringsAsFactors = F, sep = ";")
simulations <- readRDS(here::here("output", "postBurnIn_IITA_PS.rds"))

RelPar <- tibble()
for(repl in 1:nrow(SimResults)){
  for(YEAR in length(SimResults$SimOutput[[repl]]$records$CrossingBlock$grmData)){
RelPar <- rbind(RelPar,
                SimResults$SimOutput[[repl]]$records$CrossingBlock$grmData[[YEAR]][SimResults$SimOutput[[repl]]$records$CrossingBlock$grmData[[YEAR]]$pop == "c",])
}
}

write.table(x = RelPar, file = here::here("output", "ParentOriginPS.csv"),
            sep = ",", quote = FALSE,
            row.names = FALSE, col.names = TRUE)
```

```{r PostBurnIn IITA PS Extra, eval = F}

.libPaths(c(.libPaths(), "/programs/R-4.1.2/library/", "/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/"))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select <- dplyr::select
mutate <- dplyr::mutate

schemeDF <- read.csv(here::here("data","baselineScheme - IITAPSExtra.csv"), 
                     header = T, stringsAsFactors = F, sep = ";")

bsp1 <- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 4, nYrsAsCandidates = 3, maxTrainingPopSize = 6620,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = "AYT",  RmStagePhen = "SDN",
                   nParents = 30, nCrosses = 400, nProgeny = 35, nClonesToNCRP = 3,
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = T,
                   nCyclesToKeepRecords = 4,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here("code","runSchemesPostBurnIn.R"))

nRepT <- 40#readRDS(here::here("output", "BurnIn_IITA.rds")) %>% nrow()
NCT <- RhpcBLASctl::get_num_procs()
nRepMap <- 8
reps <- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]


set.seed(501)

for(i in 1:max(reps)) {

burnIn_IITA_PS <- readRDS(here::here("output", "BurnIn_IITA.rds")) %>% .[reps==i,]
start <- proc.time()[3]
postBurnIn_IITA_PS <- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = "parentSelCritBLUP",
                                           selCritPipe = "productSelCritBLUP",
                                           productFunc = "productPipelinePostBurnIn",
                                           popImprovFunc = "popImprovByParentSel",
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end <- proc.time()[3]; print(paste0(floor((end - start)/60), "mins ", floor(((end - start)%%60)), "s elapsed - bspP"))

RdsName <- paste("postBurnIn_IITA_PSPlus", i, ".rds", sep = "")
saveRDS(postBurnIn_IITA_PS,file = here::here("output", RdsName))
rm(postBurnIn_IITA_PS); rm(RdsName); rm(burnIn_IITA_PS)
}
```

```
 ****** 1 to 8
 [1] "39mins 39s elapsed - bspP"

 ****** 9 to 16
 [1] "40mins 33s elapsed - bspP"

 ****** 17 to 24
 [1] "40mins 22s elapsed - bspP"

 ****** 25 to 32

 ****** 33 to 40
 [1] "47mins 3s elapsed - bspP"
 
```


joint all the Results in a same rds File
```{r Joint PostBurnIn IITA PS Extra data, eval = FALSE}
suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(80)
files <- here::here("output", dir("output")[grep(pattern = "postBurnIn_IITA_PSPlus*", dir("output"))])
SimResults <- NULL
for(i in files){
  Results <- readRDS(i)
  SimResults <- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = here::here("output", "postBurnIn_IITA_PSPlus.rds"))
rm(SimResults)
```



####  With Genomic Selection
<br />

| FunctionsArg  | FunctionName         |
| :-----------: | :------------------: |
| selCritPop    | parentSelCritGEBV    |
| selCritPipe   | productSelCritBLUP   |
| productFunc   | productPipeline      |
| popImprovFunc | popImprovByParentSel |
|               |                      |

###### Parameters Changuing in different scenarios

- 3 Yrs as Candidates - cbsu lm34 - screen BurnIn

<br />

```{r IITA GS 3Yrs Cand, eval = F}
.libPaths(c(.libPaths(), "/programs/R-4.1.2/library/", "/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/"))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select <- dplyr::select
mutate <- dplyr::mutate
AlphaSimHlpR::popImprovOutput

schemeDF <- read.csv(here::here("data","baselineScheme - IITA.csv"), 
                     header = T, stringsAsFactors = F, sep = ";")

bsp1 <- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 5, nYrsAsCandidates = 3, maxTrainingPopSize = 6620,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = "CET",
                   RmStagePhen = "SDN",
                   nParents = 30, nCrosses = 400, nProgeny = 35, nClonesToNCRP = 3,
                   parentsFlowering = 100,
                   TrainPopSel = "trainPopSelNew",
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = F,
                   nCyclesToKeepRecords = 5,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here("code","runSchemesPostBurnIn.R"))

nRepT <- 40#readRDS(here::here("output", "BurnIn_IITA.rds")) %>% nrow()
NCT <- RhpcBLASctl::get_num_procs()
nRepMap <- 5
reps <- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]

for(i in 1:ceiling(nRepT/nRepMap)) {

burnIn_IITA_PS <- readRDS(here::here("output", "BurnIn_IITA.rds")) %>% .[reps==i,]
start <- proc.time()[3]
postBurnIn_IITA_GS <- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = "parentSelCritGEBV",
                                           selCritPipe = "productSelCritBLUP",
                                           productFunc = "productPipelinePostBurnIn",
                                           popImprovFunc = "popImprovByParentSel",
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end <- proc.time()[3]; print(paste0(floor((end - start)/60), "mins ", floor(((end - start)%%60)), "s elapsed - bspP"))

RdsName <- paste("postBurnIn_IITA_GS_3Yrs", i, ".rds", sep = "")
saveRDS(postBurnIn_IITA_GS,file = here::here("output", RdsName))
rm(postBurnIn_IITA_GS); rm(RdsName); rm(burnIn_IITA_PS)
}
```

```
 ****** 1 to 5
 [1] "650mins 3s elapsed - bspP"
 
 ****** 6 to 10
 [1] "327mins 59s elapsed - bspP"

 ****** 11 to 15
 [1] "366mins 4s elapsed - bspP"
  
 ****** 20 to 25
 [1] "371mins 9s elapsed - bspP"
 
 ****** 26 to 30
 [1] "333mins 40s elapsed - bspP"
 
 ****** 31 to 35
 [1] "366mins 29s elapsed - bspP"
 
 ****** 36 to 40
 [1] "362mins 18s elapsed - bspP"

```

joint all the Results in a same rds File
```{r Joint Simulation Results IITA GS 3Yrs, eval = FALSE}
suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(60)
files <- here::here("output", dir("output")[grep(pattern = "postBurnIn_IITA_GS_3Yrs*", dir("output"))])
SimResults <- NULL
for(i in files){
  Results <- readRDS(i)
  SimResults <- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = "output/postBurnIn_IITA_GS_3Yrs.rds")
rm(SimResults)
```

- 2 Yrs as Candidate - CBSU lm35 - screen IITA3Yr

<br />

```{r IITA GS 2Yrs Cand, eval = F}
.libPaths(c(.libPaths(), "/programs/R-4.1.2/library/", "/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/"))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select <- dplyr::select
mutate <- dplyr::mutate

schemeDF <- read.csv(here::here("data","baselineScheme - IITA.csv"), 
                     header = T, stringsAsFactors = F, sep = ";")

bsp1 <- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 5, nYrsAsCandidates = 2, maxTrainingPopSize = 6700,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = "CET",
                   RmStagePhen = "SDN",
                   nParents = 30, nCrosses = 400, nProgeny = 35, nClonesToNCRP = 3,
                   parentsFlowering = 100,
                   TrainPopSel = "trainPopSelNew",
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = F,
                   nCyclesToKeepRecords = 5,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here("code","runSchemesPostBurnIn.R"))

nRepT <- 40#readRDS(here::here("output", "BurnIn_IITA.rds")) %>% nrow()
NCT <- RhpcBLASctl::get_num_procs()
nRepMap <- 5
reps <- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]

for(i in 5:ceiling(nRepT/nRepMap)) {

burnIn_IITA_PS <- readRDS(here::here("output", "BurnIn_IITA.rds")) %>% .[reps==i,]
start <- proc.time()[3]
postBurnIn_IITA_GS <- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = "parentSelCritGEBV",
                                           selCritPipe = "productSelCritBLUP",
                                           productFunc = "productPipelinePostBurnIn",
                                           popImprovFunc = "popImprovByParentSel",
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end <- proc.time()[3]; print(paste0(floor((end - start)/60), "mins ", floor(((end - start)%%60)), "s elapsed - bspP"))

RdsName <- paste("postBurnIn_IITA_GS_2Yrs", i, ".rds", sep = "")
saveRDS(postBurnIn_IITA_GS,file = here::here("output", RdsName))
rm(postBurnIn_IITA_GS); rm(RdsName); rm(burnIn_IITA_PS)
}
```

```
 ****** 1 to 8
  [1] "557mins 43s elapsed - bspP"

 ****** 9 to 16
  [1] "526mins 55s elapsed - bspP"
  
 ****** 17 to 24
 1] "544mins 32s elapsed - bspP"
  
 ****** 25 to 32
 [1] "575mins 20s elapsed - bspP"

 ****** 33 to 40
 [1] "576mins 29s elapsed - bspP"

```

joint all the Results in a same rds File
```{r Joint Simulations Results IITA GS 2Yrs, eval = FALSE}
suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(60)
files <- here::here("output", dir("output")[grep(pattern = "postBurnIn_IITA_GS_2Yrs*", dir("output"))])
SimResults <- NULL
for(i in files){
  Results <- readRDS(i)
  SimResults <- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = "output/postBurnIn_IITA_GS_2Yrs.rds")
rm(SimResults)
```

- 1 Yrs as Candidate - CBSU lm30 - screen IITA1Yr

<br />

```{r IITA GS 1Yr Cand, eval = F}
.libPaths(c(.libPaths(), "/programs/R-4.1.2/library/", "/home/lbraatz/R/x86_64-pc-linux-gnu-library/4.1/"))


library(devtools)
library(AlphaSimR)
library(AlphaSimHlpR)

suppressMessages(library(tidyverse))
suppressMessages(library(genomicMateSelectR))
library(dplyr); library(furrr); library(here)
library(RhpcBLASctl)

select <- dplyr::select
mutate <- dplyr::mutate

schemeDF <- read.csv(here::here("data","baselineScheme - IITA.csv"), 
                     header = T, stringsAsFactors = F, sep = ";")

bsp1 <- specifyBSP(schemeDF = schemeDF,
                   nTrainPopCycles = 5, nYrsAsCandidates = 1, maxTrainingPopSize = 7500,
                   nChr = 18, effPopSize = 200, quickHaplo = F,
                   segSites = 1650, nQTL = 1000, nSNP = 500, genVar = 1500,
                   gxeVar = NULL, gxyVar = 1500, gxlVar = 750, gxyxlVar = 300,
                   meanDD = 0.23, varDD = 0.05, relAA = 0.5,
                   stageToGenotype = "CET",
                   RmStagePhen = "SDN",
                   nParents = 30, nCrosses = 400, nProgeny = 35, nClonesToNCRP = 3,
                   parentsFlowering = 100,
                   TrainPopSel = "trainPopSelNew",
                   phenoF1toStage1 = T, errVarPreStage1 = 17500,
                   useCurrentPhenoTrain = F,
                   nCyclesToKeepRecords = 5,
                   selCritPipeAdv = selCritIID,
                   selCritPopImprov =  selCritIID)

source(here::here("code","runSchemesPostBurnIn.R"))

nRepT <- 40#readRDS(here::here("output", "BurnIn_IITA.rds")) %>% nrow()
NCT <- RhpcBLASctl::get_num_procs()
nRepMap <- 8
reps <- c(rep(1:ceiling(nRepT/nRepMap), each = nRepMap))[1:nRepT]

for(i in 4:ceiling(nRepT/nRepMap)) {

burnIn_IITA_PS <- readRDS(here::here("output", "BurnIn_IITA.rds")) %>% .[reps==i,]
start <- proc.time()[3]
postBurnIn_IITA_GS <- runSchemesPostBurnIn(simulations = burnIn_IITA_PS,
                                           newBSP = bsp1,
                                           nPostBurnInCycles = 24,
                                           selCritPop = "parentSelCritGEBV",
                                           selCritPipe = "productSelCritBLUP",
                                           productFunc = "productPipelinePostBurnIn",
                                           popImprovFunc = "popImprovByParentSel",
                                           nSimCores = nRepMap,
                                           nBLASthreads = ifelse(test = ((NCT %% nrow(burnIn_IITA_PS)) == 0),
                                                            yes = floor((NCT - 1) / nrow(burnIn_IITA_PS)),
                                                            no = floor(NCT / nrow(burnIn_IITA_PS))))
end <- proc.time()[3]; print(paste0(floor((end - start)/60), "mins ", floor(((end - start)%%60)), "s elapsed - bspP"))

RdsName <- paste("postBurnIn_IITA_GS_1Yrs", i, ".rds", sep = "")
saveRDS(postBurnIn_IITA_GS,file = here::here("output", RdsName))
rm(postBurnIn_IITA_GS); rm(RdsName); rm(burnIn_IITA_PS)
}
```


```
 ****** 1 to 8
  [1] "383mins 9s elapsed - bspP"

 ****** 9 to 16
 [1] "359mins 5s elapsed - bspP"
 
 ****** 17 to 24
 [1] "375mins 35s elapsed - bspP"
  
 ****** 25 to 32
 [1] "323mins 7s elapsed - bspP"
 
 ****** 33 to 40
 [1] "380mins 24s elapsed - bspP"
```

joint all the Results in a same rds File
```{r Joint Simulations Results IITA GS 1Yr, eval = FALSE}
suppressMessages(library(tidyverse))
RhpcBLASctl::blas_set_num_threads(80)
files <- here::here("output", dir("output")[grep(pattern = "postBurnIn_IITA_GS_1Yrs*", dir("output"))])
SimResults <- NULL
for(i in files){
  Results <- readRDS(i)
  SimResults <- rbind(SimResults, Results)  
  rm(Results)
}

saveRDS(SimResults, file = "output/postBurnIn_IITA_GS_1Yr.rds")
```


[back](VD_Optimization.html)
